raiden/blockchain_events_handler.py

```diff
@@ -245,46 +245,51 @@ def handle_channel_withdraw(raiden, event):
             data['receiver_address'],
         )
 
-        raiden.handle_state_change(withdrawn_state_change)
+        raiden.handle_state_change(withdrawn_state_change, current_block_number)
 
 
-def on_blockchain_event(raiden, event):
-    log.debug('EVENT', node=pex(raiden.address), chain_event=event)
+def on_blockchain_event(raiden, event, current_block_number):
+    log.debug(
+        'EVENT',
+        node=pex(raiden.address),
+        chain_event=event,
+        block_number=current_block_number,
+    )
 
     data = event.event_data
 
     if data['event'] == EVENT_TOKEN_ADDED:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['channel_manager_address'] = address_decoder(data['args']['channel_manager_address'])
-        handle_tokennetwork_new(raiden, event)
+        handle_tokennetwork_new(raiden, event, current_block_number)
 
     elif data['event'] == EVENT_CHANNEL_NEW:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['participant1'] = address_decoder(data['args']['participant1'])
         data['participant2'] = address_decoder(data['args']['participant2'])
-        handle_channel_new(raiden, event)
+        handle_channel_new(raiden, event, current_block_number)
 
     elif data['event'] == EVENT_CHANNEL_NEW_BALANCE:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['token_address'] = address_decoder(data['args']['token_address'])
         data['participant'] = address_decoder(data['args']['participant'])
         data['balance'] = data['args']['balance']
-        handle_channel_new_balance(raiden, event)
+        handle_channel_new_balance(raiden, event, current_block_number)
 
     elif data['event'] == EVENT_CHANNEL_CLOSED:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['closing_address'] = address_decoder(data['args']['closing_address'])
-        handle_channel_closed(raiden, event)
+        handle_channel_closed(raiden, event, current_block_number)
 
     elif data['event'] == EVENT_CHANNEL_SETTLED:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
-        handle_channel_settled(raiden, event)
+        handle_channel_settled(raiden, event, current_block_number)
 
     elif data['event'] == EVENT_CHANNEL_SECRET_REVEALED:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['receiver_address'] = address_decoder(data['args']['receiver_address'])
         data['secret'] = data['args']['secret']
-        handle_channel_withdraw(raiden, event)
+        handle_channel_withdraw(raiden, event, current_block_number)
 
     # fix for https://github.com/raiden-network/raiden/issues/1508
     # balance proof updates are handled in the linked code, so no action is needed here
```

Comment:
The `handle_*`  functions below in `on_blockchain_event2` should be updates as well

Reply:
done